doc/http_methods.rst

```diff
@@ -0,0 +1,221 @@
+
+::
+     
+     Categories(CategoryID=1714953551,CategoryName='%21%C2%9Dla%C3%92l%24',Description='hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1',Picture=binary'ac9916669fAeb2')?sap-client=500   
+     
+     {"CategoryID": "1714953551", "CategoryName": "%21%C2%9Dla%C3%92l%24", "Description": "hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1", "Picture": "YmluYXJ5J2FjOTkxNjY2OWZBZWIyJw=="}
+
+
+
+
+
+==================
+Code Documentation
+==================
+
+Adding method parameter in DirectBuilder
+----------------------------------------
+
+DirectBuilder now has an additional parameter called **method**. This accepts the users choice of HTTP method to get the fuzzed requests. The DirectBuilder *init* checks for validity and calls the *Config.fuzzer* setter to set the value in the Config object.
+
+.. code-block:: python
+
+    def __init__(self, metadata, restrictions,method):
+        if method not in ["GET","DELETE","PUT","POST"]:
+            raise ValueError("The http method value \'{}\' is invalid\nUse either GET, DELETE, PUT or POST".format(method))
+        self._queryable = QueryableEntities()
+        self._metadata_string = metadata
+        self._restrictions = restrictions
+        Config.init()
+        Config.fuzzer.http_method_enabled = method
+
+In config.py, this would be used to build the Config object, which would be looked up during query construction and check which HTTP method is set.
+
+Truncating Query Options
+------------------------
+
+For Odata queries other than **GET**, query options need to be truncated. To implement this, in fuzzer.py *Query.build_string()* would first check if *Config.fuzzer.http_method_enabled == "GET"* before generating the options and appending them. 
+The list of options are illustrated in the table above.
+
+
+Generating a Body
+-----------------
+
+For PUT and POST queries, a new element for the queries need to be generated i.e. the **body** of the request. 
+The first change is returning a tuple of *query, body* instead of just the *query* from *SingleQueryable.generate()*.
+A new function *generate_body()* is added which fetches the proprties from the metadata and calls the generator on each of them, and appends them is a dictionary.
+The dictionary is jsonified before being returned as the body. This process is skipped for **GET** and **DELETE**,and they return an empty jsonified string instead, to be compatible with the tuple returned. The changes made to the generators are described further below.
+
+
+Differentiating between PUT and POST
+------------------------------------
+
+PUT is idempotent and address single entities, whereas POST isnt idempotent and addresses multiple entities. Therefore changes are made in *DirectBuilder._append_queryable()* so that PUT avoids generating multiple entities and POST avoids generating single entities, during query generation.
```

Comment:
I am confused by the idempotency, which difference is marked, but not addressed in any way.

What exactly do you mean by difference between PUT and POST idempotency? Why is then explained only difference in Single and Multiple entity usage, but not the idempotency difference?

I mean, yes, it is the property of PUT/POST in HTTP protocol - but it is a generic property and it basically about how the target service should behave. Or does it affect the ODfuzz code in any way? I don't see it in the PR. 

If yes, pls explain futher. If no, perhaps would be better to ditch the idempotency from the docs and focus only on the differences between the single/multiple entities.

Reply:
Yes, it refers to how the service should ideally behave, which is reflected in the choice between single/multiple entities while generating payloads in ODfuzz. I'll trim it down to precisely mention single/multiple entities